Source/SwiftLintBuiltInRules/Rules/Style/TrailingClosureRule.swift

```diff
@@ -77,6 +118,39 @@ private extension TrailingClosureRule {
     }
 }
 
+private extension TrailingClosureRule {
+    final class Rewriter: ViolationsSyntaxRewriter<ConfigurationType> {
+        override func visit(_ node: FunctionCallExprSyntax) -> ExprSyntax {
+            guard node.trailingClosure == nil else { return super.visit(node) }
+
+            if configuration.onlySingleMutedParameter {
+                if let param = node.singleMutedClosureParameter,
+                let converted = node.convertToTrailingClosure() {
+                     correctionPositions.append(param.positionAfterSkippingLeadingTrivia)
+                    return ExprSyntax(converted)
+                }
+            } else if let param = node.lastDistinctClosureParameter,
+                      let converted = node.convertToTrailingClosure() {
+                correctionPositions.append(param.positionAfterSkippingLeadingTrivia)
+                return ExprSyntax(converted)
+            }
+            return super.visit(node)
+        }
+
+        override func visit(_ node: ConditionElementListSyntax) -> ConditionElementListSyntax {
+            node
+        }
+
+        override func visit(_ node: ForStmtSyntax) -> StmtSyntax {
+            if let body = rewrite(node.body).formatted().as(CodeBlockSyntax.self) {
```

Comment:
Why is the `.formatted()` needed? We should generally avoid formatting if possible. For bigger closures, it formatting might destroy the user's preferred style.

Reply:
Fixed at a967b10